pkg/handler/handler.go

```diff
@@ -113,6 +115,19 @@ func New(l *zap.Logger, baseDir string) (*mux.Router, error) {
 	router.HandleFunc("/api/v1/namespaces/{namespace}/pods/{name}/log", func(w http.ResponseWriter, r *http.Request) {
 		vars := mux.Vars(r)
 		containerName := r.URL.Query().Get("container")
+
+		if containerName == "" {
+			// User may omit the container name only when the pod has a single container
+			containerName, err = getSingleContainerPodContainerName(
+				fmt.Sprintf("http://localhost:%s", port), vars["namespace"], vars["name"])
+			if err != nil {
+				w.WriteHeader(404)
```

Comment:
Hardcoding this to 404 is not correct. I would personally make `getSingleContainerPodContainerName` return a status code